### PR TITLE
[GR-1014] Show Names can display multiple fields in hover label

### PR DESCRIPTION
**Pull request by slazicoicr**

### CODE REVIEW

File: application/dash_application/plot_builder.py [hunk 1]
```diff
@@ -134,7 +134,7 @@ def generate(title_text, sorted_data, x_fn, y_fn, axis_text, colourby, shapeby,
     if colourby == shapeby:
         name_format = lambda n: "{0}".format(n[0])
     else:
-        name_format = lambda n: "{0} {1}".format(n[0], n[1])
+        name_format = lambda n: "{0}<br>{1}".format(n[0], n[1])
```
Comment (avarsava) [inline application/dash_application/plot_builder.py:184]: \<br\> -> \<br /\>

**Review by avarsava — APPROVED** [review]

**Review by callunity — COMMENTED** [review]
Can the fields have a default order that they are displayed in? Right now if I select everything, the tissue origin and tissue type are separated by the tissue preparation line, which is unintuitive. I'd propose the following order:
```
Sample
Tissue Origin
Tissue Type
Group ID
Tissue Prep
Kit
Run
```

Since Group ID can be blank and Tissue Prep and Kit can both be "Unspecified", I think it could make sense to label those lines, ie.
```
...
Br
P
Group ID: 
Tissue Preparation: Unspecified
Kit: Unspecified
200123_A00469_0123_BCKDHELYXX
```

Since the hover text is no longer just displaying names, perhaps the dropdown label could be changed from "Show Names" to "Data Labels" or similar? It might also be worth having a "Select All Labels" button.

**Comment by avarsava** [comment]
(accidentally only looked at latest commit)

File: application/dash_application/plot_builder.py (hunk 1 above)
Comment (slazicoicr) [reply application/dash_application/plot_builder.py:184]: The `<br>` is a Plotly thing: https://help.plot.ly/adding-HTML-and-links-to-charts/#step-2-the-essentials

File: application/dash_application/views/single_lane_rna.py [hunk 2]
```diff
@@ -602,3 +605,12 @@ def all_kits_requested(click):
...
     def all_library_designs_requested(click):
         return [x for x in ALL_LIBRARY_DESIGNS]
+
+    @dash_app.callback(
+        Output(ids['show-data-labels'], 'value'),
+        [Input(ids['show-all-data-labels'], 'n_clicks')],
+        [State(ids['show-data-labels'], 'options')]
+    )
+    def all_data_labels_requested(click, avail_options):
+        if click is not None:
```
Comment (avarsava) [inline application/dash_application/views/single_lane_rna.py:615]: Why the check for `None` on click?

Comment (slazicoicr) [reply application/dash_application/views/single_lane_rna.py:615]: This callback fires when the page is loaded (`click` is None), meaning that the drop box gets filled in with all values by default. I do not understand why this does not occur with the other "Fill in All` buttons.

Comment (callunity) [reply application/dash_application/views/single_lane_rna.py:615]: ^ I also noticed this behaviour and am working on a consistent fix (that involves `if click is not None` everywhere)

**Review by callunity — APPROVED** [review]

**Pull request merged**